index.js

```diff
@@ -0,0 +1,42 @@
+const checkValidNum = (value) => {
+  if (value === '') {
+    return false
+  } else {
+    return !Number.isNaN(Number(value)) || value === null && value !== undefined
+  }
+}
```

Comment:
This is more complicated than it needs to be. When `null` is typecast to a number it becomes 0 which is what you want. `undefined` becomes `NaN` which is what you want. The only tricky one is `''` because it becomes 0. This could be

```JavaScript
const checkValidNum = (value) => !isNaN(value) && value !== ''
```